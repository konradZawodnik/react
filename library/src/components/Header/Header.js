import React from "react";
import { Link } from "react-router-dom";

class Header extends React.Component {
  state = {
    bookstoreName: "Black Books",
    clicked: true,
    textColor: "white",
    backgroundColor: "black",
  };

  handleClick = () => {
    const { clicked } = this.state;
    if (clicked) {
      this.setState({
        bookstoreName: "White Books",
        textColor: "black",
        backgroundColor: "white",
      });
    } else {
      this.setState({
        bookstoreName: "Black Books",
        textColor: "white",
        backgroundColor: "black",
      });
    }
    this.setState({
      clicked: !clicked,
    });
  };

  render() {
    const { textColor, backgroundColor, bookstoreName } = this.state;
    let headerCss = {
      color: textColor,
      backgroundColor,
    };

    return (
      <div className="row header" style={headerCss} onClick={this.handleClick}>
        <center>
          <h1>{bookstoreName}</h1>
        </center>
        <Link to="/admin">
          <button className="btn btn-info goToAdmin">
            Administrator Panel
          </button>
        </Link>
      </div>
    );
  }
}

export default Header;
