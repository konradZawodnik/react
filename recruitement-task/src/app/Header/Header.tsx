"use client";
import Image from "next/image";
import { Button } from "@/components/ui/button";
import {
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import { Label } from "@/components/ui/label";
import TCLogo from "../../../public/TClogo.png";
import { useForm, Controller } from "react-hook-form";
import { LucideBug, LucideCalendar1, LucideGlobe } from "lucide-react";

export default function Header() {
  const { control } = useForm();
  return (
    <header className="bg-[#fefefe] border flex flex-row justify-between p-6 p-16 h-full max-h-24 w-full max-w-full">
      <Image
        className="dark:invert"
        src={TCLogo}
        alt="TC Logo"
        width={191}
        height={48}
        priority
      />
      <div className="flex flex-row h-1/2">
        <Button className="bg-white border border-red-600 flex flex-row items-center justify-center h-12 rounded-lg mx-2 p-3 p-6 text-red-600 w-46">
          <LucideBug className="font-bold" width={16} height={16} />
          <Label className="cursor-pointer font-bold">Zgłoś Problem</Label>
        </Button>
        <Button className="bg-blue-600 flex flex-row items-center justify-center h-12 rounded-lg ml-4 mr-4 p-3 p-6 text-white w-44">
          <LucideCalendar1 className="font-bold" width={16} height={16} />
          <Label className="cursor-pointer font-bold">Umów Wizytę</Label>
        </Button>
        <Controller
          name="selectedLangugae"
          control={control}
          render={({ field }) => (
            <Select {...field}>
              <SelectTrigger className="h-12 text-blue-500 w-24">
                <LucideGlobe className="font-bold" width={16} height={16} />
                <SelectValue placeholder="PL" />
              </SelectTrigger>
              <SelectContent className="w-24">
                <SelectGroup>
                  <SelectItem value="PL">PL</SelectItem>
                  <SelectItem value="EN">EN</SelectItem>
                  <SelectItem value="DE">DE</SelectItem>
                </SelectGroup>
              </SelectContent>
            </Select>
          )}
        />
      </div>
    </header>
  );
}
