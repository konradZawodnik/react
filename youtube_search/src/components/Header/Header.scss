.header {
  display: flex;
  justify-content: center;
  width: 100%;
}

.logo {
  height: 100%;
  margin: -17vh 0.5vw;
  max-height: 300px;
  text-align: center;
  font-size: 1.5rem;
  font-family: Impact, "Arial Black";
  line-height: 302px;
}

.red-color {
  color: red;
}

.you {
  text-shadow: 0.02em 0.02em 0.06em rgba(0, 0, 0, 0.4);
}

.youtubeIcon {
  display: flex;
  font-size: 3.5rem;
  flex-direction: row;
  justify-content: center;
  margin-left: -6vw;
}

@media (max-width: 768px) {
  .youtubeIcon {
    display: flex;
    font-size: 3.5rem;
    flex-direction: row;
    justify-content: center;
    margin-left: 0vw;
  }
}

.youtubeIconContainer {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
