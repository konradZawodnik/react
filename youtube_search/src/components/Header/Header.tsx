import { useEffect, useState } from 'react';

import SearchBar from '../SearchBar/Searchbar';

import './Header.scss';

type HeaderProps = {
    handleFormSubmit: (termFromSearchBar?: string) => void,
}

const useMedia = (query: string) => {
    const [matches, setMatches] = useState<boolean>(false);
    useEffect(() => {
        const media = window.matchMedia(query);
        if (media.matches !== matches) {
            setMatches(media.matches);
        }
        const listener = () => {
            setMatches(media.matches);
        };
        media.addListener(listener);
        return () => media.removeListener(listener);
    }, [matches, query]);

    return matches;
}

const Header = ({ handleFormSubmit }: HeaderProps) => {
    const large = useMedia("(min-width: 768px)");
    return (
        <header className='header'>
            <div className="youtubeIcon">
                <div className="youtubeIconContainer"><i className="fa fa-youtube-play red-color" aria-hidden="true" /></div>
                {large &&
                    <div className="logo">
                        <span className="you">YouTube</span>
                    </div>}
            </div>
            <SearchBar
                handleFormSubmit={handleFormSubmit}
            />
        </header>
    );
}

export default Header;
