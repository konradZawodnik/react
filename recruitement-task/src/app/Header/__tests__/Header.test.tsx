import { render } from "@testing-library/react";
import Header from "../Header";

it("renders header unchanged", () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});
