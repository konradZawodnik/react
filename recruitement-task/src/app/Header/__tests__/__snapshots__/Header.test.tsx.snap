// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header unchanged 1`] = `
<div>
  <header
    class="bg-[#fefefe] border flex flex-row justify-between p-6 p-16 h-full max-h-24 w-full max-w-full"
  >
    <img
      alt="TC Logo"
      class="dark:invert"
      data-nimg="1"
      decoding="async"
      height="48"
      src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="191"
    />
    <div
      class="flex flex-row h-1/2"
    >
      <button
        class="gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow hover:bg-primary/90 bg-white border border-red-600 flex flex-row items-center justify-center h-12 rounded-lg mx-2 p-6 text-red-600 w-46"
      >
        <svg
          class="lucide lucide-bug font-bold"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8 2 1.88 1.88"
          />
          <path
            d="M14.12 3.88 16 2"
          />
          <path
            d="M9 7.13v-1a3.003 3.003 0 1 1 6 0v1"
          />
          <path
            d="M12 20c-3.3 0-6-2.7-6-6v-3a4 4 0 0 1 4-4h4a4 4 0 0 1 4 4v3c0 3.3-2.7 6-6 6"
          />
          <path
            d="M12 20v-9"
          />
          <path
            d="M6.53 9C4.6 8.8 3 7.1 3 5"
          />
          <path
            d="M6 13H2"
          />
          <path
            d="M3 21c0-2.1 1.7-3.9 3.8-4"
          />
          <path
            d="M20.97 5c0 2.1-1.6 3.8-3.5 4"
          />
          <path
            d="M22 13h-4"
          />
          <path
            d="M17.2 17c2.1.1 3.8 1.9 3.8 4"
          />
        </svg>
        <label
          class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-bold"
        >
          Zgłoś Problem
        </label>
      </button>
      <button
        class="gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow hover:bg-primary/90 bg-blue-600 flex flex-row items-center justify-center h-12 rounded-lg ml-4 mr-4 p-6 text-white w-44"
      >
        <svg
          class="lucide lucide-calendar1 font-bold"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 14h1v4"
          />
          <path
            d="M16 2v4"
          />
          <path
            d="M3 10h18"
          />
          <path
            d="M8 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
        </svg>
        <label
          class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer font-bold"
        >
          Umów Wizytę
        </label>
      </button>
      <button
        aria-autocomplete="none"
        aria-controls="radix-:r0:"
        aria-expanded="false"
        class="flex items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 h-12 text-blue-500 w-24"
        data-placeholder=""
        data-state="closed"
        dir="ltr"
        role="combobox"
        type="button"
      >
        <svg
          class="lucide lucide-globe font-bold"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
          />
          <path
            d="M2 12h20"
          />
        </svg>
        <span
          style="pointer-events: none;"
        >
          PL
        </span>
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-down h-4 w-4 opacity-50"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 9 6 6 6-6"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
